src/transactions/feed/queryHelper.ts

```diff
@@ -66,34 +63,41 @@ export function useFetchTransactions(): QueryHookResult {
   const address = useSelector(walletAddressSelector)
   const localCurrencyCode = useSelector(getLocalCurrencyCode)
 
-  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  // track cumulative transactions and most recent page info in one state, so
+  // that they do not become out of sync
+  const [fetchedResult, setFetchedResult] = useState<{
+    transactions: TokenTransaction[]
+    pageInfo: PageInfo | null
+  }>({
+    transactions: [],
+    pageInfo: null,
+  })
   const [fetchingMoreTransactions, setFetchingMoreTransactions] = useState(false)
-  const { transactions, addTransactions } = useDeduplicatedTransactions()
 
   // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
   const [counter, setCounter] = useState(0)
   useInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
 
-  const handleResult = (result: QueryResponse, paginatedResult: boolean) => {
-    Logger.info(TAG, `Fetched ${paginatedResult ? 'next page' : 'new'} transactions`)
+  const handleResult = (result: QueryResponse, hasPreviousPage: boolean) => {
+    Logger.info(TAG, `Fetched ${hasPreviousPage ? 'next page' : 'new'} transactions`)
 
     const returnedTransactions = result.data?.tokenTransactionsV2?.transactions
+    const returnedPageInfo = result.data?.tokenTransactionsV2?.pageInfo
 
     if (returnedTransactions?.length) {
```

Comment:
I think we have to change this...
Because if the returned transactions are empty, but it's not the last page (meaning that `hasNextPage` is true, but the blockchain-api has filtered out all the unknown transactions), we still have to update `fetchedResult.pageInfo` so when we query again we get a different page.

Reply:
oh excellent point, i can add a check for pageInfo here